extensions/data-plane/data-plane-azure-storage/src/main/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobName.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Marco Pirmo (BMW AG)
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.azure.storage;
+
+import org.eclipse.edc.util.string.StringUtils;
+
+import java.util.Optional;
+
+/**
+ * Utility class responsible for determining the name under which a file will be saved.
+ */
+public class DestinationBlobName {
+
+    private final Optional<String> folderName;
+    private final Optional<String> blobName;
```

Comment:
please do not use `Optional` as class field, it's not a good practice, wrap the field with `Optional` when you need to evaluate it (in the `resolve` method)